### ISSUE TITLE
Cannot See/Select Opus From Settings

### ISSUE THREAD

**Issue by OkyDooky** [SimpleMobileTools/Simple-Voice-Recorder#178]
So, I came here to request that the Opus codec be added to the list of options to choose from to record, but then I saw several issues that implied it has already been added.
I have this app installed on two very different devices with different Android versions and both of them show it missing.

Is this a bug? Or something I missed from the change logs?

**Comment by Aga-C** [comment]
What Android versions do you mean? Opus codec is supported on Android 10+.

**Comment by OkyDooky** [comment]
>What Android versions do you mean?

7.1.2 and 9.

>Opus codec is supported on Android 10+.

This is dependent on system libraries for that? Is it because a software encoder is not available or usable for this app on its own? Playback of and file conversion to opus files is doable on Android versions <10.

**Comment by Aga-C** [comment]
> This is dependent on system libraries for that? Is it because a software encoder is not available or usable for this app on its own? 

Yes, recording is now done via built-in MediaRecorder, and it supports Opus on Android 10+ only.

**Comment by OkyDooky** [comment]
>recording is now done via

So, this was different in the past? Thank you for clarifying.
Should I change this to be a feature request for including a software encoder for Opus? Maybe it could record in another format (like FLAC, once included) and then convert it in post?

If none of these are feasible, then maybe adding a disclaimer in the app description about what formats are (currently) supported and under what conditions (eg "Opus (only on Android versions 10 and above)" like that) would be a good solution to this problem.

**Comment by Aga-C** [comment]
>So, this was different in the past? Thank you for clarifying.
Should I change this to be a feature request for including a software encoder for Opus? Maybe it could record in another format (like FLAC, once included) and then convert it in post?

No, there wasn't such feature before. I've just wanted to emphasize that, because e.g. MP3 recording is done in a different, non-system dependent way. It could be done like you've written, and maybe it will be done, because there are already some issues raised with current OPUS support (e.g. #102, #164).

**Comment by OkyDooky** [comment]
Yaaay~! "It's doable" in any form is usually good to hear in FOSS projects, since there can be so many obstacles to achieving basic (or seemingly basic) things.
That's interesting that there are so many issues, because when I first saw them I thought that it was due to the (custom) implementation of an Opus encoder. I guess it is, but the it's system one.
Thanks for responding in a helpful way. Do you contribute code to the Simple apps or are you just very familiar with the projects? I see your name quite often when browsing through issues across all the ones I use.

**Comment by Aga-C** [comment]
> Do you contribute code to the Simple apps or are you just very familiar with the projects? I see your name quite often when browsing through issues across all the ones I use.

I just use some of those apps, and thus I like helping develop them by translating into my language or do relatively simple things in the code. And actually, OPUS codec support was added by me.

**Comment by OkyDooky** [comment]
Nice. Thanks for helping out, since that improves things for everyone else. Was adding OPUS support hard?

**Comment by Aga-C** [comment]
> Was adding OPUS support hard?

Since it's natively supported by Android, it wasn't that hard: https://github.com/SimpleMobileTools/Simple-Voice-Recorder/pull/63/files.
